testing/shared/src/test/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLoggerTestRunner.scala

```diff
@@ -48,49 +48,52 @@ class PagingSelfAwareStructuredLoggerTestRunner extends CatsEffectSuite {
-          println("Number of log entry does not match expectation")
-          println(s"loggedVec.size=${loggedVec.size}, expectedNumOfPage=$expectedNumOfPage")
-          println(s"loggedVec=$loggedVec")
-        }
-        assertEquals(loggedVec.size, expectedNumOfPage)
+            val loggedVecWithIndex = loggedVec.zip(1 to loggedVec.size)
+            val allMsgValid = loggedVecWithIndex.forall { mi =>
+              val (logMsg, pageNum) = mi
+              val ctxValid =
+                logMsg.ctx.getOrElse("log_split_id", "").matches(uuidPatternRegex) &&
+                  (logMsg.ctx.getOrElse("page_size", "0 Kib").dropRight(4).toInt == pageSizeK) &&
+                  (logMsg.ctx.getOrElse("log_size", "0 Byte").dropRight(5).toInt > pageSize || expectedNumOfPage == 1)
+              if (!ctxValid) {
+                println(s"\nFailed: $suiteCaseName - $logLevel")
+                println("Logging context does not match expectation")
+                println(s"pageNum=$pageNum, logMsg.ctx=${logMsg.ctx}")
+              }
 
-        val loggedVecWithIndex = loggedVec.zip(1 to loggedVec.size)
-        val allMsgValid = loggedVecWithIndex.forall { mi =>
-          val (logMsg, pageNum) = mi
-          val ctxValid =
-            logMsg.ctx.getOrElse("log_split_id", "").matches(uuidPatternRegex) &&
-              (logMsg.ctx.getOrElse("page_size", "0 Kib").dropRight(4).toInt == pageSizeK) &&
-              (logMsg.ctx.getOrElse("log_size", "0 Byte").dropRight(5).toInt > pageSize || expectedNumOfPage == 1)
-          if (!ctxValid) {
-            println(s"\nFailed: $suiteCaseName - $logLevel")
-            println("Logging context does not match expectation")
-            println(s"pageNum=$pageNum, logMsg.ctx=${logMsg.ctx}")
-          }
+              val msgValid = expectedNumOfPage == 1 ||
+                (pageNum == loggedVec.size || logMsg.message.length > pageSize) &&
+                  logMsg.message.endsWith(s" page_size=$pageSizeK Kib") &&
+                  logMsg.message.startsWith("Page ")
+              if (!msgValid) {
+                println(s"\nFailed: $suiteCaseName - $logLevel")
+                println("Logged message page does not match expectation")
+                println(s"pageNum=$pageNum, logMsg=$logMsg")
+              }
 
-          val msgValid = expectedNumOfPage == 1 ||
-            (pageNum == loggedVec.size || logMsg.message.length > pageSize) &&
-              logMsg.message.endsWith(s" page_size=$pageSizeK Kib") &&
-              logMsg.message.startsWith("Page ")
-          if (!msgValid) {
-            println(s"\nFailed: $suiteCaseName - $logLevel")
-            println("Logged message page does not match expectation")
-            println(s"pageNum=$pageNum, logMsg=$logMsg")
+              ctxValid && msgValid
+            }
+            assertEquals(allMsgValid, true)
           }
-
-          ctxValid && msgValid
-        }
-        assertEquals(allMsgValid, true)
-      }
+        )
+      )
```

Comment:
Getting rid of the `unsafeRunSync` in tests is good, but I don't think the `unsafeRunAsync` is quite what we want.  Assertion failures are thrown by the async callback, but not intercepted and recorded by the test framework.  When I run this with false assertions, it hangs.  We want to compose these assertions into an IO and then let MUnit run them.